bindings/python/setup.py.cmakein

```diff
@@ -13,11 +13,77 @@
 # MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 # Lesser General Public License for more details.
 
-from distutils.core import setup
-
-setup(name='libiio',
-		version='${VERSION}',
-		description='Library to use the Industrial IO devices',
-		url='https://github.com/analogdevicesinc/libiio',
-		py_modules=['iio'],
-		)
+import sys
+
+if sys.version_info[0] < 3:
+    from distutils.core import setup
+    from distutils.command.install import install
+
+    config = dict()
+else:
+    from setuptools import setup
+    from setuptools.command.install import install
+
+    config = dict(long_description_content_type="text/markdown")
+
+description = "Library for interfacing with Linux IIO devices"
+
+try:
+    with open("${CMAKE_SOURCE_DIR}/README.md", "r") as fh:
+        long_description = fh.read()
+except:
+    long_description = description
+
+
+class InstallWrapper(install):
+    """Before installing we check if the
+    libiio library is actually installed"""
+
+    def run(self):
+        self._check_libiio_installed()
+        # Run the standard PyPi copy
+        install.run(self)
+
+    def _check_libiio_installed(self):
+        from platform import system as _system
+        from ctypes import CDLL as _cdll
+        from ctypes.util import find_library
+
+        if "Windows" in _system():
+            _iiolib = "libiio.dll"
+        else:
+            # Non-windows, possibly Posix system
+            _iiolib = "iio"
+        try:
+            _lib = _cdll(find_library(_iiolib), use_errno=True, use_last_error=True)
+            if not _lib._name:
+                raise
+        except:
```

Comment:
curios: it looks like this is a catch-all-exception handler; 
any idea if the `CDLL` could through other exception types that would be worth printing?
maybe it makes debugging easier when looking in CI logs later;;